GitUI/Editor/FileViewerInternal.cs

```diff
@@ -100,6 +111,47 @@ public bool ShowLineNumbers
 
         public void SetText(string text, Action openWithDifftool, bool isDiff)
         {
+            if (TotalNumberOfLines > 1)
+            {
+                // store the previous view position
+                _previousViewPosition.FirstLine = GetLineText(0);
+                _previousViewPosition.TotalNumberOfLines = TotalNumberOfLines;
+                _previousViewPosition.CaretPosition = TextEditor.ActiveTextAreaControl.Caret.Position;
+                _previousViewPosition.FirstVisibleLine = FirstVisibleLine;
+                _previousViewPosition.CaretVisible
```

Comment:
This looks like the only place where `CaretVisible` is set. And for the most part it is based off what the struct already "knows" bar `TextEditor.ActiveTextAreaControl.TextArea.TextView.VisibleLineCount`.
It kind of feels it would be a good candidate for a method placed in `ViewPosition` however it also becomes dependent on the order in which other properties are set, which is equally bad...